src/boot/efi/stub.c

```diff
@@ -84,6 +84,20 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
                         Print(L"Unable to add image options measurement: %r", err);
                         uefi_call_wrapper(BS->Stall, 1, 200 * 1000);
                 }
+
+        } else if (cmdline_len > 0) {
+                CHAR16 *options;
+                UINTN i;
+                options = AllocatePool(cmdline_len * sizeof(CHAR16));
+                for (i= 0; i < cmdline_len; i++)
+                        options[i] = cmdline[0];
+                err = tpm_log_event(SD_TPM_PCR,
```

Comment:
I think this should be refactored to avoid repeating the tpm_log_event() call and error handling in two different blocks, and only have different handling of the marshalling of the commandline that is being measured.

Reply:
hopefully addressed this now.